lib/src/license_detection/new_tokenizer.dart

```diff
@@ -0,0 +1,256 @@
+import 'dart:collection';
+
+import 'package:meta/meta.dart';
+import 'package:source_span/source_span.dart';
+import 'package:string_scanner/string_scanner.dart';
+
+@sealed
+class Token {
+  /// Normalized form of the text in [fileSpan].
+  final String value;
+
+  /// Denotes the token position.
+  final int tokenID;
+
+  /// Zero based line number of token.
+  final int line;
+
+  /// SourceSpan of the token.
+  final SourceSpan fileSpan;
```

Comment:
nit, this is often called just `span`, see for example: [YamlNode.span](https://pub.dev/documentation/yaml/latest/yaml/YamlNode-class.html)
```suggestion
  final SourceSpan span;
```